swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -42,12 +42,28 @@ public final class MenuBar: NSObject {
 
     if let button = statusItem.button {
       button.image = signedOutIcon
+      button.target = self
+      button.action = #selector(statusItemClicked)
     }
 
     createMenu()
     setupObservers()
   }
 
+  @objc private func statusItemClicked() {
+    // Bring our app to the foreground.
+    // Prevents the "Window ordered front from a non-active application" warning, which
+    // can cause display issues with the menu.
+    NSApp.activate(ignoringOtherApps: true)
+    DispatchQueue.main.asyncAfter(deadline: .now() + 0.05) { [weak self] in
```

Comment:
@ReactorScram This delay prevents a race condition where if we try to show the menu too quickly after activating the app, it closes right away.

Activating the app resets the menu state, so we need this workaround. On macOS 13 there is supposedly a [much nicer](https://developer.apple.com/documentation/swiftui/menubarextra) API for this.

Reply:
hm, hope that time is sufficient